src/local/room_name.rs

```diff
@@ -64,13 +70,54 @@ impl LocalRoomName {
         x.as_ref().parse()
     }
 
-    /// Creates a new room name from the given position parameters.
     #[inline]
-    pub(crate) fn from_coords(east: bool, south: bool, x_pos: i32, y_pos: i32) -> Self {
-        LocalRoomName {
-            x_coord: if east { x_pos } else { -x_pos - 1 },
-            y_coord: if south { y_pos } else { -y_pos - 1 },
+    pub(crate) fn from_packed(packed: u16) -> Self {
+        LocalRoomName { packed }
+    }
+
+    /// Creates a new room name from room coords with direction implicit in
```

Comment:
Just learned that `cargo doc --include-private` is a thing that can be very helpful for us!

Reply:
It's nice!